client/libra-dev/src/transaction.rs

```diff
@@ -343,6 +361,32 @@ pub unsafe extern "C" fn libra_LibraSignedTransaction_from(
                             value: *amount,
                             address: addr.into(),
                             auth_key_prefix: auth_key_prefix_buffer,
+                            metadata_bytes: vec![].as_ptr(),
+                            metadata_len: 0,
+                        },
+                    });
+                } else {
+                    update_last_error("Fail to decode transaction payload".to_string());
+                    return LibraStatus::InternalError;
+                }
+            }
+            "peer_to_peer_with_metadata_transaction" => {
```

Comment:
Should we remove support for `peer_to_peer_transaction` (maybe just kill that script altogether) and always use `with_metadata`? The two are identical except that `peer_to_peer_transaction` fills in an empty metadata for you.

Reply:
Sweet. I will remove the script after you land this.

Reply:
it requires a bit more than this PR. For example, CLI is not using this FFI layer. I can migrate other call sites in separate PR

Reply:
@sblackshear  strongly agree!